src/amo/utils/addons.js

```diff
@@ -121,15 +122,23 @@ export const getPromotedCategory = ({
   clientApp: string,
   forBadging?: boolean,
 |}): PromotedCategoryType | null => {
-  let category = null;
-  if (addon && addon.promoted && addon.promoted.apps.includes(clientApp)) {
-    category = addon.promoted.category;
-  }
+  if (!addon?.promoted) return null;
```

Comment:
```suggestion
  if (addon.promoted.length === 0) {
    return null;
  }
```

Reply:
The definition of getPromotedCategory has
`addon: AddonType | CollectionAddonType | SuggestionType | null | void,`
This makes Typescript mad

Reply:
ha, I see